diff --git a/src/Controller/ViewsBulkOperationsController.php b/src/Controller/ViewsBulkOperationsController.php
index 8a013b1..e6e58b1 100644
--- a/src/Controller/ViewsBulkOperationsController.php
+++ b/src/Controller/ViewsBulkOperationsController.php
@@ -116,17 +116,21 @@ class ViewsBulkOperationsController extends ControllerBase implements ContainerI
 
     switch ($parameters['op']) {
       case 'add':
-        foreach ($parameters['list'] as $bulkFormKey) {
-          if (!isset($tempstore_data['list'][$bulkFormKey])) {
-            $tempstore_data['list'][$bulkFormKey] = $this->getListItem($bulkFormKey);
+        if (!empty($parameters['list'])) {
+          foreach ($parameters['list'] as $bulkFormKey) {
+            if (!isset($tempstore_data['list'][$bulkFormKey])) {
+              $tempstore_data['list'][$bulkFormKey] = $this->getListItem($bulkFormKey);
+            }
           }
         }
         break;
 
       case 'remove':
-        foreach ($parameters['list'] as $bulkFormKey) {
-          if (isset($tempstore_data['list'][$bulkFormKey])) {
-            unset($tempstore_data['list'][$bulkFormKey]);
+        if (!empty($parameters['list'])) {
+          foreach ($parameters['list'] as $bulkFormKey) {
+            if (isset($tempstore_data['list'][$bulkFormKey])) {
+              unset($tempstore_data['list'][$bulkFormKey]);
+            }
           }
         }
         break;
